sonic-chassisd/scripts/chassisd

```diff
@@ -0,0 +1,332 @@
+            return
+
+        if (admin_state == MODULE_ADMIN_DOWN) or (admin_state == MODULE_ADMIN_UP):
+            #Setting the module to administratively up/down state
+            (self.log_info("Changing module {} to admin {} state".format(key,
+             'DOWN' if admin_state == MODULE_ADMIN_DOWN else 'UP')))
+            try_get(self.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
+
+#
+# Module Updater ==============================================================
+#
+class ModuleUpdater(logger.Logger):
+
+    def __init__(self, log_identifier, chassis):
+       """
+       Constructor for ModuleUpdater
+       :param chassis: Object representing a platform chassis
+       """
+       super(ModuleUpdater, self).__init__(log_identifier)
+
+       self.chassis = chassis
+       self.num_modules = chassis.get_num_modules()
+       # Connect to STATE_DB and create chassis info tables
+       state_db = daemon_base.db_connect("STATE_DB")
+       self.chassis_table = swsscommon.Table(state_db, CHASSIS_INFO_TABLE)
+       self.module_table = swsscommon.Table(state_db, CHASSIS_MODULE_INFO_TABLE)
+       self.info_dict_keys = [CHASSIS_MODULE_INFO_NAME_FIELD,
+                              CHASSIS_MODULE_INFO_DESC_FIELD,
+                              CHASSIS_MODULE_INFO_SLOT_FIELD,
+                              CHASSIS_MODULE_INFO_OPERSTATUS_FIELD]
+
+    def deinit(self):
+        """
+        Destructor of ModuleUpdater
+        :return:
+        """
+        # Delete all the information from DB and then exit
+        for module_index in range(0, self.num_modules):
+            name = try_get(self.chassis.get_module(module_index).get_name)
+            self.module_table._del(name)
+
+        if self.chassis_table is not None:
+            self.chassis_table._del(CHASSIS_INFO_KEY_TEMPLATE.format(1))
+
+    def modules_num_update(self):
+        #Check if module list is populated
+        num_modules = self.chassis.get_num_modules()
+        if num_modules == 0:
+            self.log_error("Chassisd has no modules available")
+            return
+
+        # Post number-of-modules info to STATE_DB
+        fvs = swsscommon.FieldValuePairs([(CHASSIS_INFO_CARD_NUM_FIELD, str(num_modules))])
+        self.chassis_table.set(CHASSIS_INFO_KEY_TEMPLATE.format(1), fvs)
+
+    def module_db_update(self):
+        for module_index in range(0, self.num_modules):
+            module_info_dict = self._get_module_info(module_index)
+            if module_info_dict is not None:
+                key = module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD]
+
+                if not key.startswith(ModuleBase.MODULE_TYPE_SUPERVISOR) and \
+                   not key.startswith(ModuleBase.MODULE_TYPE_LINE) and \
+                   not key.startswith(ModuleBase.MODULE_TYPE_FABRIC):
+                    (self.log_error("Incorrect module-name {}. Should start with {} or {} or {}".format(key,
```

Comment:
Is there a reason for the whole expression to be in parenthesis?
I've seen other occurrences of this pattern in this review.

Reply:
The log line was too long. So, just for the new line I put it in the parenthesis. What is the recommended approach. Using '\'?

Reply:
I'm not sure what's the policy in this package but I would just not go to the line.
The `\` is only needed in a few corner cases, it doesn't apply here.

Reply:
You are right. Cleaned up all instances in this file.